src/3d/mesh/qgsmesh3dentity_p.cpp

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+                         qgsmesh3dentity.cpp
+                         -------------------------
+    begin                : january 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmesh3dentity_p.h"
+
+#include <QOpenGLContext>
+#include <qopenglfunctions_3_2_core.h>
+
+#include <Qt3DRender/QTexture>
+#include <Qt3DRender/QParameter>
+#include <Qt3DExtras/QTextureMaterial>
+#include <Qt3DExtras/QPhongMaterial>
+#include <Qt3DExtras/QMetalRoughMaterial>
```

Comment:
it seems like some of these #includes are not needed anymore (e.g. opengl context / functions, materials from qt3d extras)